gzip_io.ml

```diff
@@ -36,6 +36,23 @@ let string s =
   IO.nwrite out (Bytes.unsafe_of_string s); (* IO wrong type *)
   IO.close_out out
 
+let string_lwt ?(chunk_size = 3000) ?(yield = Lwt.pause) s =
+  let out = output (IO.output_string ()) in
+  let b = Bytes.unsafe_of_string s in
+  let len = Bytes.length b in
+  let rec loop offset =
+    let written = 
+      let len_to_write = Int.min chunk_size (len - offset) in
+      IO.output out b offset len_to_write in
+    if offset + written >= len then Lwt.return_unit
+    else (
+      (* Yield after processing a chunk *)
+      let%lwt () = yield () in
```

Comment:
With a buffer of size 1024 we can compress maybe around 30000 chunks/s, which means yielding every ~30 us. At this point the overhead of returning to the event loop could start to become measurable. So yielding after processing around say 4096 bytes could be better. Probably not a big deal though.

(This depends on the cost of the switch, which I guesstimated at a few us...)